internal/runbits/legacy/projectmigration/projectmigration.go

```diff
@@ -1,185 +1,84 @@
 package projectmigration
 
 import (
-	"bytes"
 	_ "embed"
-	"errors"
-	"io/fs"
-	"os"
 	"path/filepath"
-	"regexp"
-	"runtime"
 
 	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/fileutils"
 	"github.com/ActiveState/cli/internal/locale"
-	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/internal/output"
-	"github.com/ActiveState/cli/internal/prompt"
-	"github.com/ActiveState/cli/pkg/localcommit"
-	"github.com/ActiveState/cli/pkg/projectfile"
-	"gopkg.in/yaml.v3"
+	"github.com/ActiveState/cli/internal/rtutils/ptr"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
 )
 
-//go:embed migrate.mac.bash
-var migrateMacScript []byte
-
-//go:embed migrate.nix.bash
-var migrateNixScript []byte
-
-//go:embed migrate.win.ps
-var migrateWinScript []byte
-
 type projecter interface {
-	Source() *projectfile.Project
 	Dir() string
-	URL() string
-	Path() string
 	LegacyCommitID() string
-	StripLegacyCommitID() error
+	SetLegacyCommit(commitID string) error
 }
 
-var prompter prompt.Prompter
-var out output.Outputer
-
-// Register exists to avoid boilerplate in passing prompt and out to every caller of
-// commitmediator.Get() for retrieving legacy commitId from activestate.yaml.
-// This is an anti-pattern and is only used to make this legacy feature palatable.
-func Register(prompter_ prompt.Prompter, out_ output.Outputer) {
-	prompter = prompter_
-	out = out_
+type migrator struct {
+	out  output.Outputer
+	proj projecter
 }
 
-func PromptAndMigrate(proj projecter) error {
-	if prompter == nil || out == nil {
-		return errs.New("projectmigration.Register() has not been called")
-	}
+func New(out output.Outputer, pj projecter) *migrator {
+	return &migrator{out: out, proj: pj}
+}
 
-	if os.Getenv(constants.DisableProjectMigrationPrompt) == "true" {
+// setupProject will ensure that the stored project matches the path that was requested. In most cases this should match,
+// and setting up a new project instance will be rare.
+func (m *migrator) setupProject(pjpath string) error {
+	if !ptr.IsNil(m.proj) && m.proj.Dir() == pjpath {
```

Comment:
Nit: we could probably just test `m.proj != nil` and then drop the `ptr` package import.

Reply:
That's what I had before, but it doesn't work because it is an interface. So `m.proj == nil` will assert if the interface is nil, it does not assert that the underlying type that satisfies the interface is nil.

Reply:
Oh thanks, I learned something new!